components/main-chef-wrapper/platform-lib/environment_darwin.go

```diff
@@ -0,0 +1,162 @@
+package platform_lib
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/chef/chef-workstation/components/main-chef-wrapper/lib"
+	"gopkg.in/yaml.v2"
+)
+
+type EnvInfo struct {
+	ChefWorkstation ChefWorkstationInfo `yaml:"Chef Workstation"`
+	Ruby            RubyInfo            `yaml:"Ruby"`
+	Path            []string            `yaml:"Path"`
+}
+
+type ChefWorkstationInfo struct {
+	Version          string               `yaml:"Version"`
+	Home             string               `yaml:"Home"`
+	InstallDirectory string               `yaml:"Install Directory"`
+	PolicyfileConfig PolicyFileConfigInfo `yaml:"Policyfile Config"`
+}
+type RubyInfo struct {
+	Executable string  `yaml:"Executable"`
+	Version    string  `yaml:"Version"`
+	RubyGems   GemInfo `yaml:"RubyGems"`
+}
+
+type GemInfo struct {
+	RubyGemsVersion   string             `yaml:"RubyGems Version"`
+	RubyGemsPlatforms []interface{}      `yaml:"RubyGems Platforms"`
+	GemEnvironment    GemEnvironmentInfo `yaml:"Gem Environment"`
+}
+
+type PolicyFileConfigInfo struct {
+	CachePath   string `yaml:"Cache Path"`
+	StoragePath string `yaml:"Storage Path"`
+}
+
+type GemEnvironmentInfo struct {
+	GemRoot string   `yaml:"Gem Root"`
+	GemHome string   `yaml:"Gem Home"`
+	GemPath []string `yaml:"Gem Path"`
+}
+
+func RunEnvironment() error {
+	// call all the environment info here.
+	envObj := WorkstationEnvInfo()
+	ymldump, err := yaml.Marshal(envObj)
+	if err != nil {
+		log.Fatalf("error: %v", err)
+	}
+	fmt.Printf("----:\n%s\n\n", string(ymldump))
```

Comment:
if you are using log then remove fmt

Reply:
log was giving timestamp(e.g--` 2022/02/08 11:11:35`), so fmt was more suitable.